rfd/0051-x11-forwarding.md

```diff
@@ -0,0 +1,269 @@
+
+New `tsh ssh` flags and ssh options:
+ - `-X`/`--x11-untrusted`: requests `untrusted` X11 forwarding for the session.
+ - `-Y`/`--x11-trusted`: requests `trusted` X11 forwarding for the session.
+ - `--x11-untrusted-timeout=<duration>`: can be used with untrusted X11 forwarding to set
+   a timeout, after which the xauth cookie will expire and the server will reject any new requests.
+
+Additionally, we will support the following OpenSSH flags for user's who want the same UX as OpenSSH:
+ - `-oForwardX11Trusted=<yes/no>`: sets the X11 forwarding trust mode.
+   - When set to `yes`, X11 forwarding will always be in `trusted` mode, whether through the
+     `-X`, `-Y`, or `-oForwardX11=yes` flag.
+   - When set to `no`, normal behavior will be used, meaning `-X` and `-Y` will lead to `untrusted`
+     and `trusted` forwarding respectively. `-oForwardX11=yes` will lead to `untrusted` forwarding.
+ - `-oForwardX11=yes`: requests X11 forwarding for the session.
+   - When used, `oForwardX11Trusted` will default to `yes` unless explicitly set.
+ - `-oForwardX11Timeout=<int>`: same effect as `--x11-timeout`.
+
+#### Untrusted as default mode
+
+In OpenSSH, the X11 forwarding flags and options have very unintuitive UX. Essentially, in
+many OpenSSH distributions, both `-X` and `-Y` will start X11 forwarding in `trusted` mode, 
+unless the client explicitly sets `ForwardX11Trusted=no` in their ssh client config. In my
+experience and research, this leads to many users becoming confused, not knowing if they are
+using `trusted` or `untrusted`, and having no idea what the difference is.
+
+The reason for this strange UX behavior is that the X Security Extension had quite a contentious 
+release. On top of the security benefits being confusing and unclear to many users, using the
+extension initially led to many X programs to perform poorly or even crash, as mentioned in the
+previous section.
+
+It seems that this UX decision is ultimately the result of trying to include an incompatible
+yet important security fix without having any effect on existing users of X11 forwarding.
+In other words the X Security Extension is an opt-in feature in OpenSSH X11 forwarding as a
+result of its issues.
+
+Two decades later, I believe we have no reason to follow the same UX and should instead learn
+from the mistakes of OpenSSH. As a security centered product, the security implications of a 
+user's actions should be as clear and unambiguous as possible. 
+
+From the get-go, `tsh ssh -X` and `tsh ssh -Y` will always lead  to `untrusted` or `trusted`
+forwarding respectively. If a user runs into an X program which still doesn't support the X 
+Security Extension, I would prefer that they experience the crash, come to customer support
+or other sources to discover the reason, and personally decide whether to switch to `trusted`
+forwarding or keep their X Server secure.
+
+This does leave us with the unfortunate fact that the UX of `tsh` and `ssh` will be different
```

Comment:
If this became a problem, we could technically add some code to detect if `tsh` is aliased and being invoked as `ssh`, and supporting the `ssh` flags only in that case.

Right?